### ISSUE TITLE
SQLite: attempt to write a readonly database

### ISSUE THREAD

**Issue by vadimkerr** [grafana/oncall#1960]
I'm not able to install the plugin when following the steps from https://github.com/grafana/oncall#getting-started.

See the following Slack threads for more info:
- https://grafana.slack.com/archives/C02LSUUSE2G/p1684301198822339
- https://grafana.slack.com/archives/C02LSUUSE2G/p1682482655294859

I was able to reproduce this, here's a full log for `oncall-engine`:

```
[uWSGI] getting INI configuration from uwsgi.ini
[log-encoder] registered format ${strftime:%Y-%m-%d %H:%M:%S} ${msgnl}
2023-05-17 16:09:46 *** Starting uWSGI 2.0.21 (64bit) on [Wed May 17 16:09:45 2023] ***
2023-05-17 16:09:46 compiled with version: 8.3.0 on 15 May 2023 18:47:22
2023-05-17 16:09:46 os: Linux-5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022
2023-05-17 16:09:46 nodename: 886176c5222b
2023-05-17 16:09:46 machine: aarch64
2023-05-17 16:09:46 clock source: unix
2023-05-17 16:09:46 detected number of CPU cores: 4
2023-05-17 16:09:46 current working directory: /etc/app
2023-05-17 16:09:46 writing pidfile to /tmp/project-master.pid
2023-05-17 16:09:46 detected binary path: /usr/local/bin/uwsgi
2023-05-17 16:09:46 !!! no internal routing support, rebuild with pcre support !!!
2023-05-17 16:09:46 uWSGI running as root, you can use --uid/--gid/--chroot options
2023-05-17 16:09:46 setgid() to 2000
2023-05-17 16:09:46 setuid() to 1000
2023-05-17 16:09:46 chdir() to /etc/app
2023-05-17 16:09:46 your memory page size is 4096 bytes
2023-05-17 16:09:46 detected max file descriptor number: 1048576
2023-05-17 16:09:46 lock engine: pthread robust mutexes
2023-05-17 16:09:46 thunder lock: disabled (you can enable it with --thunder-lock)
2023-05-17 16:09:46 uWSGI http bound on 0.0.0.0:8080 fd 7
2023-05-17 16:09:46 uwsgi socket 0 bound to TCP address 127.0.0.1:41273 (port auto-assigned) fd 6
2023-05-17 16:09:46 Python version: 3.11.3 (main, May  3 2023, 07:58:26) [GCC 8.3.0]
2023-05-17 16:09:46 Python main interpreter initialized at 0xffffae803bf0
2023-05-17 16:09:46 python threads support enabled
2023-05-17 16:09:46 your server socket listen backlog is limited to 100 connections
2023-05-17 16:09:46 your mercy for graceful operations on workers is 60 seconds
2023-05-17 16:09:46 mapped 855210 bytes (835 KB) for 5 cores
2023-05-17 16:09:46 *** Operational MODE: preforking ***
2023-05-17 16:09:46 WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0xffffae803bf0 pid: 7 (default app)
2023-05-17 16:09:46 *** uWSGI is running in multiple interpreter mode ***
2023-05-17 16:09:46 spawned uWSGI master process (pid: 7)
2023-05-17 16:09:46 spawned uWSGI worker 1 (pid: 8, cores: 1)
2023-05-17 16:09:46 spawned uWSGI worker 2 (pid: 9, cores: 1)
2023-05-17 16:09:46 spawned uWSGI worker 3 (pid: 10, cores: 1)
2023-05-17 16:09:46 spawned uWSGI worker 4 (pid: 11, cores: 1)
2023-05-17 16:09:46 spawned uWSGI worker 5 (pid: 12, cores: 1)
2023-05-17 16:09:46 spawned uWSGI http 1 (pid: 13)
2023-05-17 16:10:41 source=engine:app google_trace_id=none logger=root outbound latency=0.0260502089986403 status=200 method=HEAD url=http://grafana:3000/api/org slow=0
2023-05-17 16:10:41 source=engine:app google_trace_id=none logger=apps.grafana_plugin.helpers.client Error connecting to api instance 404 Client Error: Not Found for url: http://grafana:3000/api/access-control/users/permissions/search?actionPrefix=grafana-oncall-app
2023-05-17 16:10:41 source=engine:app google_trace_id=none logger=root outbound latency=0.007426042000588495 status=404 method=HEAD url=http://grafana:3000/api/access-control/users/permissions/search?actionPrefix=grafana-oncall-app slow=0
2023-05-17 16:10:41 source=engine:app google_trace_id=none logger=django.request Internal Server Error: /api/internal/v1/plugin/self-hosted/install
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: attempt to write a readonly database

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/app/apps/grafana_plugin/views/self_hosted_install.py", line 55, in post
    organization = Organization.objects.create(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/app/apps/user_management/models/organization.py", line 48, in create
    instance = super().create(**kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 453, in create
    obj.save(force_insert=True, using=self.db)
  File "/usr/local/lib/python3.11/site-packages/django/db/models/base.py", line 739, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/usr/local/lib/python3.11/site-packages/django/db/models/base.py", line 776, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/base.py", line 881, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/base.py", line 919, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 1270, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1416, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 79, in _execute
    with self.db.wrap_database_errors:
  File "/usr/local/lib/python3.11/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: attempt to write a readonly database
2023-05-17 16:10:41 source=engine:app google_trace_id=none logger=root inbound latency=5.412402 status=500 method=POST path=/api/internal/v1/plugin/self-hosted/install content-length=0 slow=1
2023-05-17 16:10:41 source=engine:uwsgi status=500 method=POST path=/api/internal/v1/plugin/self-hosted/install latency=5.424850 google_trace_id=- protocol=HTTP/1.1 resp_size=329 req_body_size=0
```

**Comment by Dieterbe** [comment]
I have the same issue. workaround:
```
make init start COMPOSE_PROFILES=mysql,engine,grafana,redis
```

**Comment by iskhakov** [comment]
I was able to reproduce this issue on Mac OS by running
```bash
sudo chown root engine/oncall.db
```

When I change the ownership to the user I use to run docker:
```
sudo chown $USER engine/oncall.db
```
this error is gone

**Comment by Dieterbe** [comment]
The problem is that inside the container, the userid and groupid of the file are the same as on the host.
For example on the host, my id's are `uid=1011(dieter) gid=1000(dieter)`, which are mapped literally in the docker container. inside the container, the wsgi process runs as user 1000, which is 'other' and doesn't have write permission
```
~/c/grafana-oncall ❯❯❯ docker exec -it 0704ed7b4e88 bash                                                                                                                                                      
root@0704ed7b4e88:/etc/app# ls -alh /var/lib/oncall/oncall.db /etc/app/oncall.db
-rw-r--r-- 1 1011 1000 1.5M May 17 15:30 /etc/app/oncall.db
-rw-r--r-- 1 1011 1000 1.5M May 17 15:30 /var/lib/oncall/oncall.db
root@3409a4db212f:/etc/app# apt-get update && apt-get install procps
(...)
root@3409a4db212f:/etc/app# ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   2388  1408 ?        Ss   06:56   0:00 sh -c uwsgi --disable-logging --py-autoreload 3 --ini uwsgi.ini
1000           7  0.2  0.3 148288 122904 ?       S    06:56   0:00 uwsgi --disable-logging --py-autoreload 3 --ini uwsgi.ini
1000           8  0.1  0.3 222036 103280 ?       Sl   06:56   0:00 uwsgi --disable-logging --py-autoreload 3 --ini uwsgi.ini
1000          10  0.1  0.3 222036 103280 ?       Sl   06:56   0:00 uwsgi --disable-logging --py-autoreload 3 --ini uwsgi.ini
1000          12  0.2  0.3 222036 103152 ?       Sl   06:56   0:00 uwsgi --disable-logging --py-autoreload 3 --ini uwsgi.ini
1000          14  0.2  0.3 222036 103152 ?       Sl   06:56   0:00 uwsgi --disable-logging --py-autoreload 3 --ini uwsgi.ini
1000          16  0.2  0.3 237868 124656 ?       Sl   06:56   0:00 uwsgi --disable-logging --py-autoreload 3 --ini uwsgi.ini
1000          18  0.3 26.0 8536896 8489756 ?     S    06:56   0:01 uwsgi --disable-logging --py-autoreload 3 --ini uwsgi.ini
root          19  0.0  0.0   5752  3328 pts/0    Ss   06:59   0:00 bash
root         332  0.0  0.0   9392  2944 pts/0    R+   07:02   0:00 ps aux
```

Probably something special is going on to keep the host id's, because i don't think this is standard docker behavior

**Comment by prawitdtgo** [comment]
I was workaround
`docker exec -ti grafana-oncall-engine-1 sh`
`chown root:root /var/lib/oncall`
`chmod 777 -R /var/lib/oncall`
and config plugin with http://engine:8080 again

**Comment by vadimkerr** [comment]
The issue should be fixed in [v1.2.27](https://github.com/grafana/oncall/releases/tag/v1.2.27). 
If you're using [hobby setup](https://github.com/grafana/oncall#getting-started) and seeing this error in earlier versions (or after updating to v1.2.27 from earlier versions), here's a command that should resolve permission issues:

```bash
docker-compose exec engine bash -c "chown -R 1000:2000 /var/lib/oncall"
```
